### PR TITLE
Zapier tutorial

### PR DESCRIPTION
**Pull request by Srushtika**
This PR contains the tutorial text for the IoT <> Zapier project, with links to all resources.

### CODE REVIEW

File: content/tutorials/realtime-attendance-system-iot.textile [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+MFRC522 - Huzzah Pins
+* RST  - 4
+* SDA   - 5
+* MOSI - 13 
+* MISO - 12 
+* SCK  - 14 
+* 3V - 3.3V
+* GND - GND
+
+Caution: If you try to connect the power source of the RFID card to 5V instead of 3.3V, it might fry up.
```
Comment (thisisjofrank) [inline content/tutorials/realtime-attendance-system-iot.textile]: Caution: Do not connect the power source of the RFID card reader to the 5V power output (BATT or USB) as it might fry the reader. Be sure to connect it to 3.3V.

File: content/tutorials/realtime-attendance-system-iot.textile [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+h2(#step3-mqtt-ably). Step 3 - Code to publish messages to Ably over MQTT
+
+In this step, we'll start writing the code to publish messages to Ably via MQTT when the RFID scanner detects an access card.
```
Comment (thisisjofrank) [inline content/tutorials/realtime-attendance-system-iot.textile]: In this step, we'll start writing the code to publish messages to Ably via MQTT when the RFID sensor detects an access card.

Comment (thisisjofrank) [inline content/tutorials/realtime-attendance-system-iot.textile]: In this step, we'll start writing the code to publish messages to Ably via MQTT when the RFID sensor detects an access card.

File: content/tutorials/realtime-attendance-system-iot.textile [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+  <img src="/images/tutorials/iot-attendance/boards-manager.png" style="width: 100%" alt="Boards Manager">
+</a>
+
+Next, you need to include the libraries needed for this app. You can do this by going into Arduino's 'Sketch' menu. 
+
+Sketch -> Include library -> Manage libraries. 
+
+Search for @MFRC522@ and install it. You need this library to enable the communication between the IoT board and the RFID sensor. This library comes with the @SPI@ library as well that you need to set up a serial communication protocol to communicate with any peripheral devices, in this case the RFID sensor. 
+
+The last thing that's needed is an MQTT library. You can use one of the libraries from Arduino's library manager, but for this tutorial we'll use the one by Joel Gaehwiler. You can "download ZIP library":https://www.arduinolibraries.info/libraries/mqtt directly and then include it the usual way Sketch -> Include library -> Add .ZIP library. All these libraries come with a few example sketches that you check out. We'll write our app from scratch.
```
Comment (thisisjofrank) [inline content/tutorials/realtime-attendance-system-iot.textile]: You can "download the ZIP library"

include it in the usual way:

File: content/tutorials/realtime-attendance-system-iot.textile [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+Next, instantiate the libraries so they can be used in the code. Add the following:
+
+```[cplusplus]
+MFRC522 mfrc522(SS_PIN, RST_PIN);
+WiFiClientSecure net;
+MQTTClient client;
+```
+
+Now, inside the default @setup()@ method of your sketch, add the following:
```
Comment (thisisjofrank) [inline content/tutorials/realtime-attendance-system-iot.textile]: is it worth explaining a little what setup() does? (not sure if this is outside of the scope of the tutuorial) maybe it could be a comment in the code?

File: content/tutorials/realtime-attendance-system-iot.textile [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+      byte nib1 = (array[i] >> 4) & 0x0F;
+      byte nib2 = (array[i] >> 0) & 0x0F;
+      buffer[i*2+0] = nib1  < 0xA ? '0' + nib1  : 'A' + nib1  - 0xA;
+      buffer[i*2+1] = nib2  < 0xA ? '0' + nib2  : 'A' + nib2  - 0xA;
+    }
+    buffer[len*2] = '\0';
+}
+```
+
+Every access card can be identified by a unique identifier, called UID. As you can see, in the above code, I've just included a simple UID check with a specific UID. In practice, you can have a switch case to match the UIDs with the employee names.
```
Comment (thisisjofrank) [inline content/tutorials/realtime-attendance-system-iot.textile]: remove 'just'

File: content/tutorials/realtime-attendance-system-iot.textile [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+Go ahead and verify that the sketch has no errors by clicking on the 'tick' button in the Arduino IDE. Assuming everything is good, connect your board to your computer and upload the sketch to it by clicking on the 'right arrow' button. Open the serial monitor (search icon button on the top right) to check the logs.
```
Comment (thisisjofrank) [inline content/tutorials/realtime-attendance-system-iot.textile]: connect your board to your computer with its USB cable

File: content/tutorials/realtime-attendance-system-iot.textile [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+    }
+    buffer[len*2] = '\0';
+}
+```
+
+Every access card can be identified by a unique identifier, called UID. As you can see, in the above code, I've just included a simple UID check with a specific UID. In practice, you can have a switch case to match the UIDs with the employee names.
+
+Go ahead and verify that the sketch has no errors by clicking on the 'tick' button in the Arduino IDE. Assuming everything is good, connect your board to your computer and upload the sketch to it by clicking on the 'right arrow' button. Open the serial monitor (search icon button on the top right) to check the logs. 
+
+If you have a connection error for the MFRC522 component, it's most likely due to some loose connection. The best solution would be to solder the jump wires to the board and the RFID component.
```
Comment (thisisjofrank) [inline content/tutorials/realtime-attendance-system-iot.textile]: due to a loose connection

File: content/tutorials/realtime-attendance-system-iot.textile [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+  <img src="/images/tutorials/iot-attendance/zap-setup.png" style="width: 100%" alt="Zap set up">
+</a>
+
+The action needs to be "Catch a Raw Hook", so choose that and continue. In the next screen Zapier will give you a URL, which is the WebHooks endpoint needed by Ably, so that it can trigger that URL when a message is published on a channel. Copy that out.
+
+<a href="/images/tutorials/iot-attendance/zap-wh-endpoint.png" target="_blank">
+  <img src="/images/tutorials/iot-attendance/zap-wh-endpoint.png" style="width: 100%" alt="Zap WebHooks endpoint">
+</a>
+
+In another tab, log into your Ably account, go to the Apps Dashboard. In the "Reactor" tab, click:
```
Comment (thisisjofrank) [inline content/tutorials/realtime-attendance-system-iot.textile]: We'll have a partial for this if you want it once the IFTTT branch is merged

File: content/tutorials/realtime-attendance-system-iot.textile [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+In this final step, we'll test our application out. So upload the sketch file to your board once again and place the access card near the RFID sensor. This should now result in a message in the Slack channel that you set up as shown below. Simple and easy!
+
+<a href="/images/tutorials/iot-attendance/slack-message.png" target="_blank">
+  <img src="/images/tutorials/iot-attendance/slack-message.png" style="width: 100%" alt="Customize slack messages">
+</a>
+
+h3(#further-reading). Further reading
+
+1. Learn more about the "Arduino ecosystem":https://www.arduino.cc/
+2. If you would like to find out more about Ably Reactor features and capabilities, see the "Reactor documentation":https://www.ably.io/reactor.
```
Comment (thisisjofrank) [inline content/tutorials/realtime-attendance-system-iot.textile]: Matt tells me this isn't documentation its a landing page, so I changed it to say "Find out more about Ably reactor features and capabilityes"

File: content/tutorials/realtime-attendance-system-iot.textile (hunk 4 above)
Comment (Srushtika) [reply content/tutorials/realtime-attendance-system-iot.textile]: Sounds good, a line or two should be fine, done!

File: content/tutorials/realtime-attendance-system-iot.textile (hunk 8 above)
Comment (Srushtika) [reply content/tutorials/realtime-attendance-system-iot.textile]: Cool, will update when it's available 👍🏼

File: content/tutorials/realtime-attendance-system-iot.textile (hunk 9 above)
Comment (Srushtika) [reply content/tutorials/realtime-attendance-system-iot.textile]: Ah, good catch. Fixed!
